### PR TITLE
December drop

### PR DESCRIPTION
**Pull request by sameshai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,15 +66,23 @@ For building the driver `docker` and `GO` should be installed on the system
...
    Create an image pull secret in your cluster
 
-   1. Review and retrieve the following values for your image pull secret.
+   1. ibmcloud login to the target region
+
+   2. Run - ibmcloud cr region-set global
+
+   3. Run - ibmcloud cr login
+
+   4. Run - ibmcloud ks cluster config --cluster \<cluster-name\> admin
```
Comment (prankulmahajan) [inline README.md:75]: Can you update this to
`ibmcloud ks cluster config --cluster \<cluster-name\> --admin`

File: pkg/ibmcsidriver/controller_helper.go [hunk 2]
```diff
@@ -205,6 +222,14 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 		volume.VPCVolume.VolumeEncryptionKey = nil
 	}
 
+	// Add initialOnwer if UID/GID is given as parameter.
+	// Default will be set to 0 i.e root which even if not set will be defaulted to 0 by the VPC RIAAS
+	logger.Info("Adding initial owner...", zap.Any("uid", uid), zap.Any("gid", gid))
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/controller_helper.go:227]: Can we add an if condition to do this only if uid/gid are mentioned
like,
```
if uid != 0 || gid != 0 {
  logger.Info("Adding initial owner...", zap.Any("uid", uid), zap.Any("gid", gid))
  volume.InitialOwner = &provider.InitialOwner{
		GroupID: int64(gid),
		UserID:  int64(uid),
	}
}
```

Am saying this, because the logging info "Adding initial owner" seems to indicate that the user has mentioned the initial owner value in the sc.

**Review by prankulmahajan — COMMENTED** [review]
One case which I didnt test, was what was the maximum length of UID/GID that the user can give. If there is a limit on that, we can add a case. Just a corner case.

Otherwise the PR looks good to me. 
Thanks

File: README.md (hunk 1 above)
Comment (sameshai) [reply README.md:75]: > One case which I didnt test, was what was the maximum length of UID/GID that the user can give. If there is a limit on that, we can add a check. Just a corner case.
> 
> Otherwise the PR looks good to me. Thanks

This error will come from VPC RIAAZ if there is any issue with gid and uid. Lets think later if we need to add more validations at driver side

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: README.md [hunk 3]
```diff
@@ -122,4 +139,23 @@ Pull requests are very welcome! Make sure your patches are well tested. Ideally
...
-6. Add the test results in the PR
\ No newline at end of file
+6. Add the test results in the PR
+
+# Supported features
+
+1. Dynamic PVC/PV creation and deletion with ReadWriteMany capability
+2. Deployment creation and deletion which will mount/unmount the file storage volumes.
+3. ReplicationController creation and deletion which will mount/unmount the file storage volumes.
+4. StatefulSet creation and deletion which will mount/unmount the file storage volumes.
```
Comment (sameshai) [inline README.md:149]: Just keep PV mount and mount
PV creation and deletion with RWM access. 
Mount/Unmount of file shares to or from the VPC target.

File: README.md [hunk 4]
```diff
@@ -122,4 +139,23 @@ Pull requests are very welcome! Make sure your patches are well tested. Ideally
...
+5. Defining custom storage class by providing gid/uid will allow non-root users access to file storage volumes.
```
Comment (sameshai) [inline README.md:150]: Custom gid/uid supported via the custom storage access.

File: README.md [hunk 5]
```diff
@@ -122,4 +139,23 @@ Pull requests are very welcome! Make sure your patches are well tested. Ideally
...
+6. Static PVC/PV creation and deletion with ReadWriteMany capability.
```
Comment (sameshai) [inline README.md:151]: Remove this part

File: README.md [hunk 6]
```diff
@@ -122,4 +139,23 @@ Pull requests are very welcome! Make sure your patches are well tested. Ideally
...
+
+# Unsupported features
```
Comment (sameshai) [inline README.md:153]: Remove this unsupported features

**Review by arahamad — APPROVED** [review]
lgtm

**Review by prankulmahajan — APPROVED** [review]
